src/Aspire.Hosting/DistributedApplicationOptions.cs

```diff
@@ -50,6 +50,11 @@ public DistributedApplicationOptions()
     /// </summary>
     public bool DisableDashboard { get; set; }
 
+    /// <summary>
+    /// The time provider used to override time handling inside the app host.
+    /// </summary>
+    public TimeProvider TimeProvider { get; set; } = TimeProvider.System;
```

Comment:
This doesn't need to be an API here. Just add it to services and it can be overridden.